src/game.cpp

```diff
@@ -2819,6 +2835,38 @@ void Game::toggleHud(float *statustext_time, bool *flag)
 		client->setHighlighted(client->getHighlighted(), *flag);
 }
 
+void Game::toggleMinimap(float *statustext_time, bool *flag, bool *show_hud)
+{
+	if (*show_hud && g_settings->getBool("enable_minimap")) {
+		u16 mode = mapper->getMinimapMode();	
+		mode++;
+		if (mode>6) {
+			mode = 0;
+			*flag = false;
+			statustext = L"Minimap hidden";
+		} else if (mode == 1) {
```

Comment:
Can you create an enum in minimap.h to define those numeric values for a better reading ?
Also, i think you should use a switch instead of if/else if there and a default with the mode > 6 condition